### PR TITLE
feat(pap) : power aware plugin

### PR DESCRIPTION
**Pull request by h-w-chen**
#### What type of PR is this?
Feature

#### What this PR does / why we need it:
It introduces the new feature - power aware plugin.

Its general architecture:
* main body in sysadvisor;
* CPU frequency RAPL capping utilitity lib wrapper in external manager;
* workload evict interaction is, as an interim means, abstracted as interface of podKiller.PodKiller;

By default this plugin is enabled; in order to disable it, 
```bash
--power-aware-disabled=true
```
This plugin supports DryRun mode; to enable DryRun
```bash
--power-aware-dryrun=true
```
#### Which issue(s) this PR fixes:
NA - new feature

#### Special notes for your reviewer:
Pod Evict API is not in place yet; for interim purpose, this PR assumes podKiller.PodKiller interface (not finalized and could change), provides a dummy no-op implementation, which is for now to facilitate testing and in the future to be replaced with a real meaningful implementation.

### CODE REVIEW

File: pkg/agent/sysadvisor/plugin/poweraware/component/evictor.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+func (l loadEvictor) isBE(pod *v1.Pod) bool {
+	qosLevel, err := l.qosConfig.GetQoSLevelForPod(pod)
+	if err != nil {
+		// unknown, not BE anyway
+		return false
+	}
+
+	return qosLevel == apiconsts.PodAnnotationQoSLevelSharedCores ||
```
Comment (h-w-chen) [inline pkg/agent/sysadvisor/plugin/poweraware/component/evictor.go:60]: share-core type is NOT BE, is it?

File: pkg/agent/sysadvisor/plugin/poweraware/component/reconsiler.go
Comment (zzzzhhb) [inline pkg/agent/sysadvisor/plugin/poweraware/component/reconsiler.go:1]: reconsiler?typo?

File: pkg/agent/sysadvisor/plugin/poweraware/component/evictor.go (hunk 1 above)
Comment (zzzzhhb) [reply pkg/agent/sysadvisor/plugin/poweraware/component/evictor.go:60]: no, shared-cores is GT. The `apiconsts.PodAnnotationQoSLevelReclaimedCores`  is equal to BE.  But in katalyst, there is no concept of BE. We only have the four types of QoS, so please just use the `apiconsts.PodAnnotationQoSLevelReclaimedCores`.

Comment (h-w-chen) [reply pkg/agent/sysadvisor/plugin/poweraware/component/evictor.go:60]: thanks for verification. will change code to reflect that semantic.

File: pkg/agent/sysadvisor/plugin/poweraware/component/reconsiler.go
Comment (h-w-chen) [reply pkg/agent/sysadvisor/plugin/poweraware/component/reconsiler.go:1]: yes typo. will rectify it.

**Pull request closed without merging**